src/main/java/au/edu/wehi/idsv/SAMEvidenceSource.java

```diff
@@ -279,7 +281,24 @@ public File getSVFile() {
 	public void assertPreprocessingComplete() {
 		File svFile = getSVFile();
 		if (svFile != null && !svFile.exists()) {
-			throw new IllegalStateException(String.format("Missing required file %s. See GRIDSS pipeline examples and documentation.", svFile));
+			try {
+				// Create the hard link
+				Path existingFile = Paths.get(getFile().getPath());  // The existing file
+				Path newLink = Paths.get(svFile.getPath());        // The path for the hard link
+
+				Files.createLink(newLink, existingFile);
+				log.debug("Hard link created successfully for input file");
+
+				Path existingIndexFile = Paths.get(getFile().getPath()+".bai");  // The existing file
```

Comment:
Can it be cram and crai too?

Reply:
@ilyasoifer No, even if you give it a cram file it converts it into a bam file